docs/design/hub-and-spoke/switch-validate/switch-validate.md

```diff
@@ -5,9 +5,861 @@
+  "missing_group_buckets": [
+    {
+      "port": 4,
+      "vlan": 4,
+      "vni": 4
+    }
+  ]
+}
+```
+
+#### Extended logical port info (extendedLogicalPortInfo)
+
+```json
+{
+  "actual": {
+    "physical_ports": [
+      10,
+      11
+    ],
+    "type": "lag"
+  },
+  "expected": {
+    "physical_ports": [
+      10,
+      11,
+      13
+    ],
+    "type": "bfd"
+  },
+  "physical_ports": [
+    1
+  ],
+  "logical_port_number": 100,
+  "type": "lag"
+}
+```
+
+#### Extended meter info (extendedMeterInfo)
+
+```json
+{
+  "actual": {
+    "flags": [
+      "KBPS"
+    ],
+    "rate": 250,
+    "burst_size": 220
+  },
+  "expected": {
+    "flags": [
+      "STATS"
+    ],
+    "rate": 200,
+    "burst_size": 230
+  },
+  "flags": [
+    "STATS"
+  ],
+  "rate": 300,
+  "burst_size": 200,
+  "cookie": 10,
+  "flow_id": "10",
+  "flow_path": "e0b0716e-cc59-4ef6-8572-518bdcdc5c72",
+  "y_flow_id": "111",
+  "meter_id": 1
+}
+```
+
+#### Extended rule info (extendedRuleInfo)
+
+```json
+{
+  "cookie": -9223372036854776000,
+  "cookie_hex": "800000000000001b",
+  "table_id": 10,
+  "priority": 5,
+  "flow_id": "9",
+  "flow_path": "10a1df3a-dc30-453e-8671-246f82ea0e77",
+  "y_flow_id": "111",
+  "flags": [
+    "RESET_COUNTERS"
+  ],
+  "match": [
+    {
+      "value": 1,
+      "mask": 1,
+      "field": "IN_PORT"
+    }
+  ],
+  "instructions": {
+    "go_to_table": 30,
+    "go_to_meter": 40,
+    "write_metadata": {
+      "value": 200,
+      "mask": 200
+    },
+    "apply_actions": [
+      "SET_FIELD"
+    ],
+    "write_actions": [
+      "SET_FIELD"
+    ]
+  },
+  "expected": {
+    "table_id": 10,
+    "priority": 5,
+    "flags": [],
+    "match": [
+      {
+        "value": 9,
+        "mask": 10,
+        "field": "IN_PORT"
+      }
+    ],
+    "instructions": {
+      "go_to_table": 3,
+      "go_to_meter": 4,
+      "write_metadata": {
+        "value": 100,
+        "mask": 100
+      },
+      "apply_actions": [
+        "SET_FIELD"
+      ],
+      "write_actions": [
+        "SET_FIELD"
+      ]
+    }
+  },
+  "actual": {
+    "table_id": 1,
+    "priority": 2,
+    "flags": [
+      "RESET_COUNTERS"
+    ],
+    "match": [
+      {
+        "value": 7,
+        "mask": 8,
+        "field": "ETH_TYPE"
+      }
+    ],
+    "instructions": {
+      "go_to_table": 1,
+      "go_to_meter": 1,
+      "write_metadata": {
+        "value": 10,
+        "mask": 10
+      },
+      "apply_actions": [
+        "PORT_OUT"
+      ],
+      "write_actions": [
+        "PORT_OUT"
+      ]
+    }
+  }
+}
+```
+
+### Validate rules and meters installed on the switch
+
+`GET https://{host}/v2/switches/{switch_id}/validate`
+
+**Default: without any query params response contains extended info about groups, lags, meters and
+rules.**
+
+Response payload(`extended info for all fields`)
+
+```json
+{
+  "groups": {
+    "excess": [
+      <extendedGroupInfo>
+    ],
+    "misconfigured": [
+      <extendedGroupInfo>
```

Comment:
Can we/should we use same data format to describe missing/excess/proper and misconfigured entries? Only misconfigured entries can have expected and actual data, all othere fields only "actual". There is at least 2 approach possible. We can define expected/actual data for each entried and do not split them on top level on excess/misconfigured/missing/proper (caller can analise data in each entry and decide what kind of entry it is - we can add some markers inside entries to help called in this task). Or we can keep current approach - define excess/misconfigured/missing/proper on top level but do not put actual expected fields into their entries (except misconfigured).